paasta_tools/cleanup_kubernetes_jobs.py

```diff
@@ -95,44 +104,118 @@ def alert_state_change(application: Application, soa_dir: str) -> Generator:
         raise
 
 
+def instance_is_not_bouncing(
```

Comment:
could this function check that only one deployment has a non-zero amount of expected replicas in order to simplify the logic? (perhaps I'm missing something - in which case, maybe we could add some comments here 'cause I'm not quite sure I'm reading this right in the first place :p)